benchmarks/test_differentiation.py

```diff
@@ -0,0 +1,83 @@
+"""Benchmarks for differentiation using ProtoSym and SimpleCAS.
+
+Differentiation using ProtoSym and SimpleCAS is currently benchmarked against
+SymPy's and SymEngine's symbolic differentiation.
+
+"""
+from typing import Callable
+from typing import TypeVar
+
+import pytest
+import symengine
+import sympy
+
+from protosym import simplecas
+
+
+ExprType = TypeVar("ExprType")
+Fixture = Callable[..., ExprType]
+
+
+@pytest.mark.benchmark(group="differentiate nested sine first derivative")
+class TestNestedSineFirstDerivative:
+    """Differentiate ``sin(sin(sin(sin(sin(sin(x))))))`` w.r.t. ``x`` once."""
+
+    @staticmethod
+    def test_simplecas(benchmark: Fixture[simplecas.Expr]) -> None:
+        """Differentiate using SimpleCAS."""
+        x = simplecas.Symbol("x")
+        sin = simplecas.sin
+        expr = sin(sin(sin(sin(sin(sin(x))))))
+        result = benchmark(expr.diff, x)
+        assert result.eval_f64({x: 1.0}) == pytest.approx(0.13877489681259086)
+
+    @staticmethod
+    def test_sympy(benchmark: Fixture[sympy.Expr]) -> None:
+        """Differentiate using SymPy."""
+        x = sympy.Symbol("x")
+        sin = sympy.sin
+        expr = sin(sin(sin(sin(sin(sin(x))))))
+        result = benchmark(expr.diff, x)
+        assert result.evalf(subs={x: 1.0}) == pytest.approx(0.13877489681259086)
```

Comment:
I didn't know about `pytest.approx`

Reply:
I hadn't until recently when I was looking for a non-NumPy way to assert close.